Tests/StreamChatTests/StateLayer/MessageState_Tests.swift

```diff
@@ -0,0 +1,223 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChat
+@testable import StreamChatTestTools
+import XCTest
+
+@available(iOS 13.0, *)
+final class MessageState_Tests: XCTestCase {
+    private var channelId: ChannelId!
+    private var env: TestEnvironment!
+    private var messageId: MessageId!
+    private var testError: TestError!
+    private var unrelatedMessageId: MessageId!
+    // Main actor since message state mutates on the main actor and we want to read results on the main actor
+    @MainActor private var messageState: MessageState!
+    
+    override func setUpWithError() throws {
+        channelId = .unique
+        env = TestEnvironment()
+        messageId = .unique
+        unrelatedMessageId = .unique
+        
+        // Channel is required for saving messages
+        env.client.databaseContainer.write { session in
+            try session.saveChannel(payload: self.makeChannelPayload(messageId: nil))
+        }
+    }
+
+    @MainActor override func tearDownWithError() throws {
+        env.cleanUp()
+        channelId = nil
+        env = nil
+        messageId = nil
+        messageState = nil
+        unrelatedMessageId = nil
+    }
+
+    // MARK: - Observing Message
+    
+    func test_observingMessage_whenMessageChanges_thenStateChanges() async throws {
+        try await setUpMessageState()
+        XCTAssertEqual(nil, await messageState.message.localState)
+        
+        try await modifyMessage { dto in
+            dto.localMessageState = .sendingFailed
+        }
+        XCTAssertEqual(LocalMessageState.sendingFailed, await messageState.message.localState)
+    }
+    
+    // MARK: - Observing Reactions
+    
+    func test_restoringReactions_whenReactionsStored_thenInitialStateIsSet() async throws {
+        let messagePayload = makeMessagePayload(reactionCount: 3, messageId: messageId)
+        try await env.client.databaseContainer.write { session in
+            try session.saveMessage(payload: messagePayload, for: self.channelId, syncOwnReactions: true, cache: nil)
+            // Unrelated
+            try session.saveMessage(payload: self.makeMessagePayload(reactionCount: 3, messageId: self.unrelatedMessageId), for: self.channelId, syncOwnReactions: true, cache: nil)
```

Comment:
for methods with many params, we should move the params in a separate line:
```swift
saveMessage(
    payload: messagePayload,
    ....
)
```

Reply:
Done, reviewed other cases as well in the changed code